### ISSUE TITLE
Potential issue with msys2 and PATH

### ISSUE THREAD

**Issue by Sopel97** [glinscott/nnue-pytorch#186]
`...msys2\mingw64\bin` needs to be in PATH, but that might not be setup by default, in which case we need to modify the environment programatically

**Comment by ppigazzini** [comment]
I just installed msys2 in a sandbox with the latest official installer, the path is not updated automatically and there is not an option.
The user can add "C:\msys64\usr\bin;C:\msys64\mingw64\bin;" in the path environment variable (first folder for msys2 applications like  `grep`, second one for the mingw-w64 applications if any).
Passing an env to subprocess is an easy option but the user must be aware to update the msys2 path with a not standard msys2 installation. Chocolatey at example installs msys2 in C:\tools\msys64\

**Comment by Sopel97** [comment]
we can probably infer the path from `where gcc`, because we already require gcc in path, and the only way for newer gcc on windows appears to be msys2

**Comment by Sopel97** [comment]
According to https://superuser.com/a/480115/388191 `where` only searches `cwd` and directories from `PATH`. So we should be fine.

**Comment by ppigazzini** [comment]
In this case `where` works only if we ask the user to update the `PATH` (a viable option).

**Comment by Sopel97** [comment]
Does `where gcc` not work in the sandbox you set up? If `gcc` was not in path then the issue reported on discord would have failed during verification, much earlier.

**Comment by ppigazzini** [comment]
Check `where /?` to view all the options. It scanned all C: drive to find `grep.exe`
```
C:\Users\WDAGUtilityAccount>where /R C:\ grep.exe
C:\msys64\usr\bin\grep.exe
```


Another option: we use a `worker.cmd` launcher on windows, the user doesn't need to update the PATH 
```cmd
@echo off
set PATH=C:\tools\msys64\mingw64\bin;C:\tools\msys64\usr\bin;%PATH%
env\bin\python3.exe -i worker.py
```

**Comment by Sopel97** [comment]
Alternatively we may have to fall back to have the user provide msys2 directory as an argument, with an attempt to infer it otherwise.

**Comment by Sopel97** [comment]
> Another option: we use a worker.cmd launcher on windows, the user doesn't need to update the PATH

What do we set the path to though

**Comment by ppigazzini** [comment]
> > Another option: we use a worker.cmd launcher on windows, the user doesn't need to update the PATH
> 
> What do we set the path to though

Standard installation paths used by chocolatey, see the scripts
https://github.com/ppigazzini/worker-setup

**Comment by Sopel97** [comment]
Ah, so you basically want the script to install msys?

**Comment by ppigazzini** [comment]
Yes, the windows worker setup is meant for the CPU contributor not able to follow the installation tutorial on the wiki. Chocolatey has the plus to auto update msys2 after the installation and the minus to use a not standard installation path.
I install only the required applications (and I use `wget` + `unzip` instead that the bigger `git`) , but you are right that msys2 takes some space. Fortunately nobody has never protested.
```
# install packages if not already installed
pacman -S --noconfirm --needed unzip make mingw-w64-x86_64-gcc mingw-w64-x86_64-python3
```

**Comment by Sopel97** [comment]
Okay. I think you have a point and it would be a good idea to fallback to automatic msys installation if not provided by the user. I'll try to prepare something around tomorrow.
